tasks/create-pyxis-image/create-pyxis-image.yaml

```diff
@@ -108,10 +108,15 @@ spec:
         echo "${pyxisCert}" > /tmp/crt
         echo "${pyxisKey}" > /tmp/key
 
-        for containerImage in $(jq -r '.components[].repository' "${SNAPSHOT_SPEC_FILE}") ; do
-
-            MEDIA_TYPE=$(skopeo inspect --raw "docker://${containerImage}" | jq -r .mediaType)
-            skopeo inspect --no-tags "docker://${containerImage}" > /tmp/skopeo-inspect.json
+        COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
+        for (( i=0; i < $COMPONENTS; i++ )); do
```

Comment:
couldn't this keep the for each and be like
```
for component in $(jq -rc '.components[]' ${SNAPSHOT_SPEC_FILE}") ; do
  CONTAINER_IMAGE=$(jq -r '.containerImage' <<< $component)
  REPOSITORY=$(jq -r '.repository' <<< $component)
...
```

Reply:
I actually prefer to iterate over an index - it seems much cleaner to me. It could break easily if there was a space somewhere in there. Maybe it wouldn't and maybe there will never be a space, but it still seems cleaner and safer this way.

Reply:
To elaborate more, it's one thing to iterate over single string values, but I don't like iterating over structured jsons like this.